txstratum/prometheus.py

```diff
@@ -16,21 +26,59 @@
 METRIC_INFO: Dict[str, str] = {
     'miners_count': 'Number of connected miners',
     'total_hashrate_ghs': 'Hashrate (Gh/s)',
-    'txs_solved': 'Number of solved transactions',
     'txs_timeout': 'Number of timeouts when solving transactions',
     'blocks_found': 'Number of blocks found',
     'uptime': 'Service uptime',
     'tx_queue': 'Number of transactions in the queue',
     'block_template_error': 'Number of errors updating block template',
 }
 
+# Metrics that we update through pubsub events
+METRICS_PUBSUB = {
+    'txs_jobs_received': Counter(
+        'txs_jobs_received',
+        'Number of transactions received by the API',
+    ),
+    'txs_solved': Counter(
+        'txs_solved',
+        'Number of solved transactions',
+        labelnames=['miner_type', 'miner_address'],
+    ),
+    'txs_solved_weight': Histogram(
+        'txs_solved_weight',
+        'Txs solved histogram by tx weight',
+        buckets=(17, 18, 19, 20, 21, 22, 23, 25, float("inf")),
```

Comment:
Why no 24?

Reply:
Once I made an analysis of distribution of weights in the mainnet, and according to it we didn't have many txs in this weight zone.

So I thought we could just have a larger bucket without losing too much information.

The more buckets we have the more data we will generate in Prometheus, so I tried just creating what is necessary.

But I guess we could add the 24 if you think it's necessary, one more won't kill us.

![histogram](https://user-images.githubusercontent.com/5041650/150016119-ddfd9410-c4f4-48aa-b9ba-f9325fb7a612.png)

Reply:
I think it's fine to skip 24